src/app/RootRoute.js

```diff
@@ -79,19 +79,21 @@ export default {
             cb(null, [require('app/components/pages/Witnesses')]);
```

Comment:
again, please walk me through the use of `require` and `require.ensure` in this context

Reply:
When require is wrapped with require.ensure it will make webpack to dynamically load a component and it's deps

Reply:
Are we very sure this is a dynamic usage? commonjs `require` is synchronous, meaning that the dependency has to be available within the current tick of the event loop. Why not pull that `require` out as an ES6 `import` at the top, thus above we would have

```ES6
import WitnessComponent from 'app/components/pages/Witnesses';
```

and at line 79, we'd have
```ES6
cb(null, [WitnessComponent]);
```

My point is that this isn't working like a dynamic route loader that checks whether it's on a route like `/@someUser` and then loads the code at runtime, saving those bytes from the initial bundle and making time to first content shorter. That `require` is still being parsed out at compile time and the associated code added to the bundle. As a result, it isn't clear to me why we want the added complexity of a `require` as opposed to another standard ES6 `import`.

The tell is that CommonJS `require` is synchronous, whereas the ES6 module loader, `System`, returns a promise with the module once it's loaded.

If I'm wrong or misunderstanding any of these points, please let me know!

Reply:
It's working like a dynamic route loader (code splitting in webpack's terminology). See https://webpack.js.org/api/module-methods/#require-ensure for details
It really decreases the bundle's size, for example loading react rte editor only when user need  to compose a post allows us to save hundreds of kilobytes in the main bundle file.
To see how it works watch the Network tab and navigate to /submit.html